processor/src/test/java/org/mapstruct/ap/test/conversion/java8stream/Source.java

```diff
@@ -0,0 +1,109 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.conversion.java8stream;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * @author Filip Hrisafov
+ */
+public class Source {
+
+
+    private List<Integer> ints;
+
+    private Stream<Integer> stream;
```

Comment:
The erasure is a problem as it will allow a `List<String>` to be converted from/to `Stream<Integer>`. I think that we need to do something like in the `BuiltInMethod`(s) there the generics are handled nice.  Maybe even use them instead